arbitrum-client/src/errors.rs

```diff
@@ -12,7 +12,14 @@ pub enum ArbitrumClientError {
     /// Error thrown when serializing/deserializing calldata/retdata
     Serde(String),
     /// Error thrown when converting between relayer & smart contract types
-    ConversionError(ConversionError),
+    Conversion(ConversionError),
+    /// Error thrown when querying events
+    EventQuerying(String),
+    /// Error thrown when a commitment can't be found in the Merkle tree
+    CommitmentNotFound,
+    /// Error thrown when an update can't be found for a specific node
+    /// in the Merkle tree
+    NodeChangeNotFound,
```

Comment:
From my comment above, I don't think semantically this error should ever be thrown. If we can't find a node change event, it is implicitly the default value at that height. So unless I'm missing something we can remove this